File: OLSource1/v1/VS_officedev/codesnippet/CSharp/extending-word-documents-and-excel-workbooks-in-vsto-add-ins-at-run-time_5.cs
        private Word.Document document1 = null;

        private void CreateDocument1()
        {
            document1 = this.Application.Documents.Add(ref missing,
                ref missing, ref missing, ref missing);
            this.Application.DocumentBeforeSave += 
                new Word.ApplicationEvents4_DocumentBeforeSaveEventHandler(
                Application_DocumentBeforeSave);
        }

        private void Application_DocumentBeforeSave(Word.Document Doc, 
            ref bool SaveAsUI, ref bool Cancel)
        {
            if (Type.ReferenceEquals(Doc, document1)) 
            {
                Cancel = true;
            }
        }

File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/extending-word-documents-and-excel-workbooks-in-vsto-add-ins-at-run-time_5.vb
    Private document1 As Word.Document = Nothing

    Private Sub CreateDocument1()
        document1 = Me.Application.Documents.Add()
    End Sub

    Private Sub Application_DocumentBeforeSave(ByVal Doc As Word.Document, _
        ByRef SaveAsUI As Boolean, ByRef Cancel As Boolean) _
        Handles Application.DocumentBeforeSave
        If Type.ReferenceEquals(Doc, document1) Then
            Cancel = True
        End If
    End Sub